diff --git a/app/Listeners/RemoveMountsAfterInstall.php b/app/Listeners/RemoveMountsAfterInstall.php
new file mode 100644
--- /dev/null
+++ b/app/Listeners/RemoveMountsAfterInstall.php
@@ -0,0 +1,23 @@
+<?php
+
+namespace Pterodactyl\Listeners;
+
+use Pterodactyl\Models\Mount;
+use Pterodactyl\Models\MountServer;
+
+class RemoveMountsAfterInstall
+{
+    /**
+     * Handle the event.
+     *
+     * @param  object  $event
+     * @return void
+     */
+    public function handle($event)
+    {
+        $mounts = Mount::where('mount_on_install', '=', true)->where('auto_mount', '=', false)->get();
+        foreach ($mounts as $mount) {
+            MountServer::where('mount_id', $mount->id)->where('server_id', $event->server->id)->delete();
+        }
+    }
+}
diff --git a/app/Models/Mount.php b/app/Models/Mount.php
--- a/app/Models/Mount.php
+++ b/app/Models/Mount.php
@@ -51,6 +51,8 @@
         'id' => 'int',
         'read_only' => 'bool',
         'user_mountable' => 'bool',
+        'mount_on_install' => 'bool',
+        'auto_mount' => 'bool',
     ];
 
     /**
@@ -62,6 +64,8 @@
         'target' => 'required|string',
         'read_only' => 'sometimes|boolean',
         'user_mountable' => 'sometimes|boolean',
+        'mount_on_install' => 'sometimes|boolean',
+        'auto_mount' => 'sometimes|boolean',
     ];
 
     /**
diff --git a/app/Providers/EventServiceProvider.php b/app/Providers/EventServiceProvider.php
--- a/app/Providers/EventServiceProvider.php
+++ b/app/Providers/EventServiceProvider.php
@@ -15,14 +15,18 @@
 use Pterodactyl\Observers\EggVariableObserver;
 use Pterodactyl\Listeners\AuthenticationListener;
 use Pterodactyl\Events\Server\Installed as ServerInstalledEvent;
+use Pterodactyl\Listeners\RemoveMountsAfterInstall;
 use Pterodactyl\Notifications\ServerInstalled as ServerInstalledNotification;
 use Illuminate\Foundation\Support\Providers\EventServiceProvider as ServiceProvider;
 
 class EventServiceProvider extends ServiceProvider
 {
     /**
      * The event to listener mappings for the application.
      */
     protected $listen = [
-        ServerInstalledEvent::class => [ServerInstalledNotification::class],
+        ServerInstalledEvent::class => [
+            ServerInstalledNotification::class,
+            RemoveMountsAfterInstall::class,
+        ],
     ];
diff --git a/app/Services/Servers/ServerCreationService.php b/app/Services/Servers/ServerCreationService.php
--- a/app/Services/Servers/ServerCreationService.php
+++ b/app/Services/Servers/ServerCreationService.php
@@ -19,6 +19,8 @@
 use Pterodactyl\Repositories\Eloquent\ServerVariableRepository;
 use Pterodactyl\Services\Deployment\AllocationSelectionService;
 use Pterodactyl\Exceptions\Http\Connection\DaemonConnectionException;
+use Pterodactyl\Models\Mount;
+use Pterodactyl\Models\MountServer;
 
 class ServerCreationService
 {
@@ -154,6 +156,7 @@
             // Create the server and assign any additional allocations to it.
             $server = $this->createModel($data);
 
+            $this->storeActiveMounts($server);
             $this->storeAssignedAllocations($server, $data);
             $this->storeEggVariables($server, $eggVariableData);
 
@@ -234,6 +237,25 @@
         return $model;
     }
 
+    /**
+     *
+     */
+    private function storeActiveMounts(Server $server)
+    {
+        $mounts = Mount::where(function ($query) {
+            $query->where('mount_on_install', '=', true)
+                  ->orWhere('auto_mount', '=', true);
+        })->get();
+        foreach($mounts as $mount) {
+            $mountServer = (new MountServer())->forceFill([
+                'mount_id' => $mount->id,
+                'server_id' => $server->id,
+            ]);
+
+            $mountServer->save();
+        }
+    }
+
     /**
      * Configure the allocations assigned to this server.
      */
diff --git a/app/Services/Servers/ReinstallServerService.php b/app/Services/Servers/ReinstallServerService.php
--- a/app/Services/Servers/ReinstallServerService.php
+++ b/app/Services/Servers/ReinstallServerService.php
@@ -24,8 +24,14 @@
      */
     public function handle(Server $server): Server
     {
+        $mounts = \Pterodactyl\Models\Mount::where('mount_on_install', true)->orWhere('auto_mount', true)->get();
+        foreach($mounts as $mount) {
+            \Illuminate\Support\Facades\DB::table('mount_server')
+                ->updateOrInsert(['mount_id' => $mount->id, 'server_id' => $server->id]);
+        }
+
         return $this->connection->transaction(function () use ($server) {
             $server->fill(['status' => Server::STATUS_INSTALLING])->save();
 
             $this->daemonServerRepository->setServer($server)->reinstall();
 
             return $server->refresh();
         });
     }
 }
diff --git a/database/migrations/2022_04_03_151827_add_auto_mount_to_mounts_table.php b/database/migrations/2022_04_03_151827_add_auto_mount_to_mounts_table.php
new file mode 100644
--- /dev/null
+++ b/database/migrations/2022_04_03_151827_add_auto_mount_to_mounts_table.php
@@ -0,0 +1,33 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddAutoMountToMountsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('mounts', function (Blueprint $table) {
+            $table->boolean("mount_on_install")->after("user_mountable")->default(false);
+            $table->boolean("auto_mount")->after("mount_on_install")->default(false);
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('mounts', function (Blueprint $table) {
+            //
+        });
+    }
+}
diff --git a/resources/views/admin/mounts/view.blade.php b/resources/views/admin/mounts/view.blade.php
--- a/resources/views/admin/mounts/view.blade.php
+++ b/resources/views/admin/mounts/view.blade.php
@@ -89,6 +89,40 @@
                                 </div>
                             </div>
                         </div>
+
+                        <div class="row">
+                            <div class="form-group col-md-6">
+                                <label class="form-label">Mount on Install</label>
+
+                                <div>
+                                    <div class="radio radio-success radio-inline">
+                                        <input type="radio" id="pMountOnInstallFalse" name="mount_on_install" value="0" @if(!$mount->mount_on_install) checked @endif>
+                                        <label for="pMountOnInstallFalse">False</label>
+                                    </div>
+
+                                    <div class="radio radio-warning radio-inline">
+                                        <input type="radio" id="pMountOnInstall" name="mount_on_install" value="1" @if($mount->mount_on_install) checked @endif>
+                                        <label for="pMountOnInstall">True</label>
+                                    </div>
+                                </div>
+                            </div>
+
+                            <div class="form-group col-md-6">
+                                <label class="form-label">Auto Mount</label>
+
+                                <div>
+                                    <div class="radio radio-success radio-inline">
+                                        <input type="radio" id="pAutoMountFalse" name="auto_mount" value="0" @if(!$mount->auto_mount) checked @endif>
+                                        <label for="pAutoMountFalse">False</label>
+                                    </div>
+
+                                    <div class="radio radio-warning radio-inline">
+                                        <input type="radio" id="pAutoMount" name="auto_mount" value="1" @if($mount->auto_mount) checked @endif>
+                                        <label for="pAutoMount">True</label>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
                     </div>
 
                     <div class="box-footer">